qr_generator/lib/screens/scan_view/scan_view.dart

```diff
@@ -0,0 +1,154 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:permission_handler/permission_handler.dart';
+import 'package:qr_generator/screens/scan_view/bloc/scan_view_bloc.dart';
+import 'package:qr_generator/screens/widgets/error_view.dart';
+import 'dart:io';
+import 'package:flutter/foundation.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+
+class ScanView extends StatefulWidget {
+  ScanView({Key? key}) : super(key: key);
+
+  @override
+  _ScanViewState createState() => _ScanViewState();
+}
+
+class _ScanViewState extends State<ScanView> {
+  late ThemeData _theme;
+  late Barcode result;
+  late QRViewController controller;
+  final GlobalKey qrKey = GlobalKey(debugLabel: 'QR');
+
+  @override
+  void reassemble() {
+    super.reassemble();
+    if (Platform.isAndroid) {
+      controller.pauseCamera();
+    }
+    controller.resumeCamera();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _theme = Theme.of(context);
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Scan View'),
+        actions: [
+          IconButton(
+              onPressed: () async => await controller.pauseCamera(),
+              icon: const Icon(Icons.pause)),
+          IconButton(
+              onPressed: () async => await controller.resumeCamera(),
+              icon: const Icon(Icons.play_arrow))
+        ],
+      ),
+      body: SafeArea(
+        child: BlocListener<ScanViewBloc, ScanViewStateBloc>(
+            listener: (context, state) {
+          if (state.scanStatus == ScanStatus.loaded) {
+            _showBottomSheet(
+                context,
+                ListTile(
+                  leading: CircleAvatar(
+                      backgroundColor:
+                          state.seedValidated ? Colors.green : Colors.red,
+                      child: Icon(
+                        state.seedValidated ? Icons.check : Icons.close,
+                        color: Colors.white,
+                      )),
+                  title: Text(state.seedValidated ? 'Valid' : 'Invalid'),
+                  subtitle: Text(state.seedData),
+                ));
+          }
+        }, child: BlocBuilder<ScanViewBloc, ScanViewStateBloc>(
+          builder: (context, state) {
+            if (state.scanStatus == ScanStatus.loaded ||
+                state.scanStatus == ScanStatus.initial) {
+              return BlocProvider.value(
+                value: BlocProvider.of<ScanViewBloc>(context),
+                child: buildQrView(context),
```

Comment:
**fix**: don't use functions to return widgets, that's not good. Prefer `const` constructors and widgets instead! See [here](https://stackoverflow.com/a/53234826) and [here](https://api.flutter.dev/flutter/widgets/StatelessWidget-class.html) read the "Performance considerations" section

Reply:
Done!